teuthology/orchestra/cluster.py

```diff
@@ -144,3 +144,13 @@ def exclude(self, *roles):
             if remote not in matches.remotes:
                 c.add(remote, has_roles)
         return c
+
+    def filter(self, func):
+        """
+        Return a cluster whose remotes are filtered by `func`.
```

Comment:
Maybe giving here a usage example is a good idea.

Reply:
Added